eureka2-write-server/src/main/java/com/netflix/eureka2/server/channel/SenderReplicationChannel.java

```diff
@@ -43,11 +43,8 @@ public SenderReplicationChannel(TransportClient client, ReplicationChannelMetric
 
     @Override
     public Observable<ReplicationHelloReply> hello(final ReplicationHello hello) {
-        if (state.get() == STATE.Closed) {
-            return Observable.error(CHANNEL_CLOSED_EXCEPTION);
-        }
-        if (!state.compareAndSet(STATE.Idle, STATE.Handshake)) {
-            return Observable.error(HANDSHAKE_FINISHED_EXCEPTION);
+        if (!moveToState(STATE.Idle, STATE.Handshake)) {
```

Comment:
Is it possible for the handshake processing to fail? If so, this channel will be stuck in state handshake and have no transition out. We should either let handshake -> handshake to be valid or better yet, close this channel if the handshake fails so we can retry.